Feat: AI Request Transformer plugin (#632)
Kong/developer.konghq.com: app/_kong_plugins/ai-request-transformer/index.md

categories:
  - ai

tags:
  - ai
  - ai-gateway

search_aliases:
  - ai
  - llm
  - artificial
  - intelligence
  - language
  - model

related_resources:
  - text: Transform a request body using OpenAI

>     url: /how-to/transform-a-client-request-with-ai/

Reviewer: ```suggestion
    url: /how-to/transform-a-client-request-with-ai/
  - text: AI Response Transformer plugin
    url: /plugins/ai-response-transformer/
```

---

The AI Request Transformer plugin uses a configured LLM service to transform a client request body before proxying the request upstream.

> This plugin supports `llm/v1/chat` requests for the same providers as the [AI Proxy plugin](/plugins/ai-proxy/).

Reviewer: ```suggestion
This plugin supports the same `llm/v1/chat` requests and providers as the [AI Proxy plugin](/plugins/ai-proxy/).
```

It also uses the same configuration and tuning parameters as the AI Proxy plugin, under the [`config.llm`](/plugins/ai-request-transformer/reference/#schema--config-llm) block.

The AI Request Transformer plugin runs **before** all of the [AI prompt](/plugins/?terms=ai%2520prompt) plugins and the
AI Proxy plugin, allowing it to also transform requests before sending them to a different LLM.

## How it works

{% include plugins/ai-transformer-diagram.md %}

> 1. The {{site.base_gateway}} admin sets up an `llm` configuration block, following the same 
> [configuration format](/plugins/ai-proxy/reference/) as the AI Proxy plugin, 
> and the same `driver` capabilities.

Reviewer: ```suggestion
1. The {{site.base_gateway}} admin sets up an [`llm` configuration block](/plugins/ai-request-transformer/reference/#schema--config-llm).
```
